app/components/satis/input_array/component.html.slim

```diff
@@ -0,0 +1,41 @@
+div.satis-input-array data-controller="satis-input-array" data-satis-input-array-input-name-value="#{options[:sub_attribute].present? ? "#{form.object_name}[#{attribute.to_s}][][#{options[:sub_attribute]}]" : "#{form.object_name}[#{attribute.to_s}][]"}"
+    / loops incorrect number of time, some unknown issue
+    / - options[:collection].each do |field|
+    /     = form.fields_for "#{attribute.to_s}[]", field do | attr_form |
+    /         = attr_form.text_field options[:sub_attribute], class: 'form-control mt-1'
+
+
+    / works correctly for array of objects
+    / - options[:collection].each do |field|
+    /     = fields_for "#{form.object_name}[#{attribute.to_s}][]", field do | attr_form |
+    /         = attr_form.text_field options[:sub_attribute], class: 'form-control mt-1'
+
+    template data-satis-input-array-target="rowTemplate"
+        .flex data-satis-input-array-target="inputRow"
+            div.flex-auto
+                = text_field_tag  nil, nil, class: 'form-control mt-1', data: { action: "input->satis-input-array#input" }
+            div.flex-none.shrink-0
+                button.w-10.h-full.items-center data-action="click->satis-input-array#removeRow"
+                    i.fas.fa-xmark
+
+    div data-satis-input-array-target="rowsContainer"
+        - if options[:sub_attribute].present? 
```

Comment:
For sub_attribute, is the intent to indicate what the value should be, or what the value attribute is? Perhaps we should call it value_method in line with this:

https://github.com/rails/rails/blob/8015c2c2cf5c8718449677570f372ceb01318a32/actionview/lib/action_view/helpers/form_options_helper.rb#L400

Reply:
sub_attribute renamed to value_method